packages/custom-objects-importer/src/main.js

```diff
@@ -0,0 +1,295 @@
+    // $FlowFixMe
+    const { body: { results: existingObjects } } = await this.client.execute(
+      existingObjectsRequest
+    )
+
+    const requestsList = batches.map(
+      (newObjects: Array<CustomObjectDraft>): Array<ClientRequest> =>
+        this._createOrUpdateObjects(existingObjects, newObjects)
+    )
+
+    const functionsList = requestsList.map(
+      (requests: Array<ClientRequest>): Function =>
+        this._createPromiseReturningFunction(requests)
+    )
+
+    return CustomObjectsImporter.promiseMapSerially(functionsList)
+  }
+
+  _createPromiseReturningFunction(requests: Array<ClientRequest>): Function {
+    return async (): Promise<void> => {
+      await Promise.all(
+        requests.map((request: ClientRequest): ExecutionResult =>
+          this._executeCreateOrUpdateAction(request)
+        )
+      )
+    }
+  }
+
+  _executeCreateOrUpdateAction(request: ClientRequest): ExecutionResult {
+    let update
+    if (request.body && request.body.update) update = request.body.update
+
+    return this.client
+      .execute(request)
+      .then((): ExecutionResult => {
+        if (update) {
+          this._summary.updated += 1
+        } else {
+          this._summary.created += 1
+        }
+        return Promise.resolve()
+      })
+      .catch((error: Error): ExecutionResult => {
+        this._summary.errors.push(error.message || error)
+        let msg
+        if (this.continueOnProblems) {
+          if (update) {
+            this._summary.updateErrorCount += 1
+            msg = 'Update error occurred but ignored. See summary for details'
+          } else {
+            this._summary.createErrorCount += 1
+            msg = 'Create error occurred but ignored. See summary for details'
+          }
+          this.logger.error(msg)
+          return Promise.resolve()
+        }
+        if (update) {
+          this._summary.updateErrorCount += 1
+          msg =
+            'Process stopped due to error while updating custom object. See summary for details'
+        } else {
+          this._summary.createErrorCount += 1
+          msg =
+            'Process stopped due to error while creating custom object. See summary for details'
+        }
+        this.logger.error(msg)
+
+        return Promise.reject(
+          new CustomObjectImportError(
+            msg,
+            this._summary,
+            error.message || error
+          )
+        )
+      })
+  }
+
+  _createOrUpdateObjects(
+    existingObjects: Array<CustomObject>,
+    newObjects: Array<CustomObjectDraft>
+  ): Array<ClientRequest> {
+    this.logger.info('Executing...')
+
+    const createOrUpdateObjects = newObjects.map(
+      (newObject: CustomObjectDraft): CustomObjectDraft | null => {
+        const existing = existingObjects.find(
+          (oldObject: CustomObject): boolean =>
+            oldObject.key === newObject.key &&
+            oldObject.container === newObject.container
```

Comment:
why wouldn't you use `isEqual` here?
you do elsewhere already

Reply:
I opted for this solution for performance reasons, since `value` in [custom objects](https://docs.commercetools.com/http-api-projects-custom-objects.html#customobject) can hold any sort of data and potentially be really huge it could be really draining to map over the whole object with `.find()` instead of quickly checking the `key` and `container` values.

Reply:
I actually meant

```js
isEqual(old.key, new.key) && isEqual(old.container, new.container)
```

but sure